include/class.task.php

```diff
@@ -1366,11 +1374,77 @@ static function create($vars=false) {
             $task->assign($form, $_errors);
         }
 
+        $task->onNewTask();
+
         Signal::send('task.created', $task);
 
         return $task;
     }
 
+    function onNewTask($vars=array()) {
+        global $cfg, $thisstaff;
+
+        if (!$cfg->alertONNewTask() // Check if alert is enabled
+            || !($dept=$this->getDept())
+            || ($dept->isGroupMembershipEnabled() == Dept::ALERTS_DISABLED)
+            || !($email=$cfg->getAlertEmail())
+            || !($tpl = $dept->getTemplate())
+            || !($msg=$tpl->getNewTaskAlertMsgTemplate())
+        ) {
+            return;
+        }
+
+        // Check if Dept recipients is Admin Only
+        $adminOnly = ($dept->isGroupMembershipEnabled() == Dept::ALERTS_ADMIN_ONLY);
+
+        // Alert recipients
+        $recipients = array();
+
+        // Department Manager
+        if ($cfg->alertDeptManagerONNewTask()
+            && $dept->getManagerId()
+            && !$adminOnly)
+            $recipients[] = $dept->getManager();
+
+        // Department Members
+        if ($cfg->alertDeptMembersONNewTask() && !$adminOnly)
+            foreach ($dept->getMembersForAlerts() as $M)
+                $recipients[] = $M;
+
+        $options = array();
+        $staffId = $thisstaff ? $thisstaff->getId() : 0;
+
+        $msg = $this->replaceVars($msg->asArray(), $vars);
+
+        $sentlist=array();
+        foreach ($recipients as $k=>$staff) {
+            if (!is_object($staff)
+                // Don't bother vacationing staff.
+                || !$staff->isAvailable()
+                // No need to alert the poster!
+                || $staffId == $staff->getId()
+                // No duplicates.
+                || isset($sentlist[$staff->getEmail()])
+                // Make sure staff has access to task
+                || !$this->checkStaffPerm($staff)
+            ) {
+                continue;
+            }
+            $alert = $this->replaceVars($msg, array('recipient' => $staff));
+            $email->sendAlert($staff, $alert['subj'], $alert['body'], null, $options);
+            $sentlist[$staff->getEmail()] = 1;
+        }
+
+        // Alert Admin ONLY if not already a staff
+        if ($cfg->alertAdminONNewTask()
+                && !in_array($cfg->getAdminEmail(), $sentlist)) {
+                //&& ($dept->isGroupMembershipEnabled() != Dept::ALERTS_DISABLED)) {
```

Comment:
Looks like you can go ahead and get rid of this since you did the check on line 1389

Reply:
lol that was embarrassing.. thank you 🤣